
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>From Mascot to sequence by hand - Proteomic-Genomic Nexus (AP11)</title>
  <meta name="author" content="Intersect and Ramaciotti Institute (UNSW)">

  
  <meta name="description" content="Working with reverse strand proteins requires special attention, as the position arithmetic and logic needs
to be worked out with care. As developers &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://IntersectAustralia.github.com/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand">
  <link href="/ap11/favicon.png" rel="icon">
  <link href="/ap11/stylesheets/bootstrap/bootstrap.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/ap11/stylesheets/bootstrap/responsive.css" media="screen, projection" rel="stylesheet" type="text/css">
  <link href="/ap11/stylesheets/syntax/syntax.css" media="screen, projection" rel="stylesheet" type="text/css">
  <style type="text/css">
    body {
      padding-bottom: 40px;
    }
    h1 {
      margin-bottom: 15px;
    }
    img {
      max-width: 100%;
    }
    .sharing, .meta, .pager {
      margin: 20px 0px 20px 0px;
    }
    .page-footer p {
      text-align: center;
    }
  </style>
  <script src="/ap11/javascripts/libs/jquery.js"></script>
  <script src="/ap11/javascripts/libs/modernizr-2.0.js"></script>
  <script src="/ap11/javascripts/libs/bootstrap.js"></script>
  <script src="/ap11/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/ap11/atom.xml" rel="alternate" title="Proteomic-Genomic Nexus (AP11)" type="application/atom+xml">
  
  

</head>

<body   >
  <nav role="navigation"><div class="navbar">
  <div class="navbar-inner">
    <div class="container">
      <a class="btn btn-navbar" data-toggle="collapse" data-target=".nav-collapse">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </a>

      <a class="brand" href="/ap11/">Proteomic-Genomic Nexus (AP11)</a>

      <div class="nav-collapse">
        <ul class="nav">
  <li><a href="/ap11/">Blog</a></li>
  <li><a href="/ap11/blog/archives">Archives</a></li>
</ul>


        <ul class="nav pull-right" data-subscription="rss">
          <li><a href="/ap11/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
          
        </ul>

        
          <form class="pull-right navbar-search" action="http://google.com/search" method="get">
            <fieldset role="search">
              <input type="hidden" name="q" value="site:IntersectAustralia.github.com/ap11" />
              <input class="search-query" type="text" name="q" results="0" placeholder="Search"/>
            </fieldset>
          </form>
        
      </div>
    </div>
  </div>
</div>
</nav>
  <div class="container">
    <div class="row-fluid">
      
<article class="hentry span9" role="article">

  
  <header class="page-header">
    
      <h1 class="entry-title">From Mascot to Sequence by Hand</h1>
    
    
      <p class="meta">
        








  


<time datetime="2012-09-13T17:24:00+10:00" pubdate data-updated="true">Sep 13<span>th</span>, 2012</time>
        
      </p>
    
  </header>


<div class="entry-content"><p>Working with reverse strand proteins requires special attention, as the position arithmetic and logic needs
to be worked out with care. As developers, we need to have a documented procedure to be able to check
amino acid and nucleotide sequences and this post is an attempt in that direction.</p>

<!-- more -->


<p>First, given an <em>accession</em> table and mascot results (in DAT) format, the following bash command
outputs proteins of interest in the mascot search</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
</pre></td><td class='code'><pre><code class='bash'><span class='line'>awk <span class="s1">&#39;$1 ~ /C$/ &amp;&amp; $3 ~ /_YEAST$/{print $3}&#39;</span> accession.txt | <span class="k">while </span><span class="nb">read </span>ploc; <span class="k">do </span>grep <span class="s2">&quot;$ploc&quot;</span> ETD_exclusion_with_inclusion_120517_Band14.txt ; <span class="k">done</span>
</span></code></pre></td></tr></table></div></figure>


<p>We get lines as the followings:</p>

<figure class='code'><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
<span class='line-number'>6</span>
<span class='line-number'>7</span>
<span class='line-number'>8</span>
<span class='line-number'>9</span>
<span class='line-number'>10</span>
<span class='line-number'>11</span>
<span class='line-number'>12</span>
<span class='line-number'>13</span>
</pre></td><td class='code'><pre><code class='text'><span class='line'>...
</span><span class='line'>q94_p3=0,755.392593,0.000455,6,IDQPAGR,51,000000000,24.30,0000000000000001020,0,0;&quot;YP264_YEAST&quot;:0:300:306:1
</span><span class='line'>q187_p8=0,803.377319,-0.000271,2,ASDIENR,6,000000000,12.48,0000002000000001000,0,0;&quot;YPR46_YEAST&quot;:0:79:85:1
</span><span class='line'>q317_p2=1,864.437469,-0.003421,4,AMKSNEK,8,005000070,11.04,0000002000000000020,0,0;&quot;YRB30_YEAST&quot;:0:182:188:2
</span><span class='line'>q356_p6=0,880.429031,0.000417,4,GNDVYSAK,24,0000000030,13.38,0000000000000002010,0,0;&quot;APN1_YEAST&quot;:0:48:55:1,&quot;VPS10_TALSN&quot;:0:118:125:1
</span><span class='line'>q1253_p1=2,1286.609390,0.000658,5,KTKDMSGGWK,33,040300600030,12.57,0000000001000002010,0,0;&quot;ARB1_YEAST&quot;:0:223:232:3
</span><span class='line'>q1253_p3=2,1286.609390,0.000658,5,KTKDMSGGWK,33,070000600030,11.03,0000000001000002010,0,0;&quot;ARB1_YEAST&quot;:0:223:232:3
</span><span class='line'>&quot;ARB1_YEAST&quot;=68334.57,&quot;ABC transporter ATP-binding protein ARB1 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) GN=ARB1 PE=1 SV=1&quot;
</span><span class='line'>q492_p1=2,943.473938,-0.002490,4,SSKVTKSK,8,0060000000,18.26,0000000002000000000,0,0;&quot;ARN1_YEAST&quot;:0:180:187:1,&quot;LPXD_RICM5&quot;:0:330:337:1
</span><span class='line'>q492_p2=2,943.473938,-0.002490,4,SSKVTKSK,8,0600000000,18.26,0000000002000000000,0,0;&quot;ARN1_YEAST&quot;:0:180:187:1,&quot;LPXD_RICM5&quot;:0:330:337:1
</span><span class='line'>q554_p4=0,967.473953,0.002495,5,TIVDTIAR,46,0000060000,21.85,0000001001000002010,0,0;&quot;ARX1_YEAST&quot;:0:225:232:1
</span><span class='line'>q19_p9=1,703.386429,-0.000781,4,NEVKSK,18,00000000,13.94,0000000000100002000,0,0;&quot;ATG26_YEAST&quot;:0:144:149:1,&quot;ILVD_SULDN&quot;:0:534:539:1,&quot;LOC1_DEBHA&quot;:0:163:168:1
</span><span class='line'>...
</span></code></pre></td></tr></table></div></figure>


<p>Let&#8217;s pick a protein match there <strong>ARB1_YEAST</strong>. We search for that protein in <a href="http://www.uniprot.org/">UniProtKB</a>
and we find it as shown in image below</p>

<p><a href="/ap11/files/protein-search-1.png"><img src="/ap11/files/protein-search-1.png" width="550"></a></p>

<p>We see there the locus name <strong>YER036C</strong>, which we would find in the <em>accession.txt</em> if we looked for our protein of interest. In the
UniProtKB there is a <strong>cross-refs</strong> link which we click. It just points to one of the sections at the bottom of the page as shown here</p>

<p><a href="/ap11/files/protein-search-2.png"><img src="/ap11/files/protein-search-2.png" width="550"></a></p>

<p>We follow the link to the EMBL sequence database, particularly the translation code shown. Follow that in new window as it is another
website, <a href="http://www.ebi.ac.uk/">EMBL-EBI</a>, particularly the page describing the sequence for that protein</p>

<p><a href="/ap11/files/protein-search-3.png"><img src="/ap11/files/protein-search-3.png" width="550"></a></p>

<p>as the preview above should confirm. There we just go to sequence (see red arrow), a link to the bottom, where we see
the sequence as displayed in here</p>

<p><a href="/ap11/files/protein-search-4.png"><img src="/ap11/files/protein-search-4.png" width="550"></a></p>

<p>So, just click the &#8220;Show full sequence&#8221; link (JavaScript) to reveal the full sequence if it is too long. Voila, the amino acid sequence!</p>

<p>Bear in mind that this whole exercise assumes a reference genome. You can see it just below the locus name in the first screenshot above. Hope
this instructions help to navigate the complex web of proteomic/genomic data.</p>
</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Intersect and Ramaciotti Institute (UNSW)</span></span>

      








  


<time datetime="2012-09-13T17:24:00+10:00" pubdate data-updated="true">Sep 13<span>th</span>, 2012</time>
      


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://IntersectAustralia.github.com/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand/" data-via="" data-counturl="http://IntersectAustralia.github.com/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand/" >Tweet</a>
  
  
  <div class="g-plusone" data-size="medium"></div>
  
  
    <div class="fb-like" data-send="true" data-width="450" data-show-faces="false"></div>
  
</div>

    
    
    <section>
      <h1>Comments</h1>
      <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
    </section>
    
    <ul class="pager">
      
      <li class="previous"><a class="basic-alignment left"
        href="/ap11/blog/2012/08/29/notes-from-demo-1/" title="Previous Post:
        notes from demo 1">&laquo; notes from demo 1</a></li>
      
      <li><a href="/ap11/blog/archives">Blog Archives</a></li>
      
      <li class="next"><a class="basic-alignment right" href="/ap11/blog/2012/09/13/notes-from-demo-2/"
        title="Next Post: Notes from demo 2">Notes from demo 2
        &raquo;</a></li>
      
    </ul>
  </footer>
</article>

<aside class="sidebar-nav span3">
  
    <section class="well">
  <ul id="recent_posts" class="nav nav-list">
    <li class="nav-header">Recent Posts</li>
    
      <li class="post">
        <a href="/ap11/blog/2013/04/30/final-product/">Final product</a>
      </li>
    
      <li class="post">
        <a href="/ap11/blog/2012/09/13/notes-from-demo-2/">Notes from demo 2</a>
      </li>
    
      <li class="post">
        <a href="/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand/">From Mascot to sequence by hand</a>
      </li>
    
      <li class="post">
        <a href="/ap11/blog/2012/08/29/notes-from-demo-1/">notes from demo 1</a>
      </li>
    
      <li class="post">
        <a href="/ap11/blog/2012/08/22/proteomic-genomic-nexus-presentation/">proteomic-genomic nexus presentation</a>
      </li>
    
  </ul>
</section>

<section class="well">
  <ul id="recent_posts" class="nav nav-list">
    <li class="nav-header">Tags</li>
  </ul>
  <ul id="recent_posts" class="nav nav-pills">
    
      <li>
        <a href="/ap11/tags/protgenexus">protgenexus</a>
      </li>
    
      <li>
        <a href="/ap11/tags/proteomics">proteomics</a>
      </li>
    
      <li>
        <a href="/ap11/tags/genomics">genomics</a>
      </li>
    
      <li>
        <a href="/ap11/tags/transcriptome">transcriptome</a>
      </li>
    
      <li>
        <a href="/ap11/tags/DIISRTE">DIISRTE</a>
      </li>
    
      <li>
        <a href="/ap11/tags/fundedByAustralianNationalDataService">fundedByAustralianNationalDataService</a>
      </li>
    
      <li>
        <a href="/ap11/tags/andsApps">andsApps</a>
      </li>
    
      <li>
        <a href="/ap11/tags/ap11">ap11</a>
      </li>
    
      <li>
        <a href="/ap11/tags/sprint demo">sprint demo</a>
      </li>
    
  </ul>
</section>









  
</aside>


    </div>
  </div>
  <footer role="contentinfo" class="page-footer"><hr/>
<div class="span8 offset1">
   <a href="http://www.ands.org.au/"><img id="ands-logo" src="http://www.ands.org.au/assets/images/ands-full-b.png" style="border: 1px solid gray; float:left; display:block;margin:.5em;"></a>
   &nbsp;<br/>This project is supported by the Australian National Data Service (ANDS). ANDS is supported by the
   Australian Government through the National Collaborative Research Infrastructure Strategy Program and 
   the Education Investment Fund (EIF) Super Science Initiative.
</div>
<div class="span8 offset1">
   <a rel="license" href="http://creativecommons.org/licenses/by/3.0/au/"><img alt="Creative Commons License" id="cc-logo" src="http://i.creativecommons.org/l/by/3.0/au/88x31.png" style="display:block;float:left;margin:.5em;"></a>
   This <span xmlns:dct="http://purl.org/dc/terms/" href="http://purl.org/dc/dcmitype/Text" rel="dct:type">site</span> by 
   <a xmlns:cc="http://creativecommons.org/ns#" href="http://www.intersect.org.au" property="cc:attributionName" rel="cc:attributionURL">Intersect</a> and the 
   <a xmlns:cc="http://creativecommongs.org/ns#" href="http://www.ramaciotti.unsw.edu.au/" property="cc:attributionName" rel="cc:attributionURL">Ramaciotti Centre</a>,
   <a href="http://www.unsw.edu.au">University of New South Wales</a> is licensed under a 
   <a rel="license" href="http://creativecommons.org/licenses/by/3.0/au/">Creative Commons Attribution 3.0 Australia License</a>. &copy; 2013
   <i>(Powered by <a href="http://octopress.org">Octopress</a>)</i>
</div>

</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'intersectaustralia';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://IntersectAustralia.github.com/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand/';
        var disqus_url = 'http://IntersectAustralia.github.com/ap11/blog/2012/09/13/from-mascot-to-sequence-by-hand/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = 'http://' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>



<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
